zingolib/src/config.rs

```diff
@@ -98,6 +98,45 @@ pub fn load_clientconfig(
     Ok(config)
 }
 
+/// Same as load_clientconfig but doesn't panic when the server can't be reached
+pub fn load_clientconfig_serverless(
+    lightwallet_uri: http::Uri,
+    data_dir: Option<PathBuf>,
+    chain: ChainType,
+    monitor_mempool: bool,
+) -> std::io::Result<ZingoConfig> {
+    use std::net::ToSocketAddrs;
+    match format!(
+        "{}:{}",
+        lightwallet_uri.host().unwrap(),
```

Comment:
Who's passing this uri to this function?   Should we add a error enum to handle the failures that can occur inside this fn?

Probably using all the functionality that `thiserror` offers to be ergonomic in the definition?

Reply:
Given that the parameter is a Uri, how can host() and port() fail?

Can we write unit-tests that force the exercise of these failure modes?

Reply:
The ideal thing would be to alter the `construct_lightwalletd_uri` method to return a `Result<http::Uri, http::uri::InvalidUri>`. We could use `panic::catch_unwind` meanwhile in tests, and then have another PR with the refactor in place.

Reply:
I agree let's change the return of `construct_lightwalletd_uri` to the `Result` suggested.